examples/example-migration-configuration/src/main/java/org/technologybrewery/baton/ReplaceRegexMatchUsingUtilsMigration.java

```diff
@@ -0,0 +1,41 @@
+package org.technologybrewery.baton;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.technologybrewery.baton.util.FileUtils;
+
+import java.io.File;
+
+public class ReplaceRegexMatchUsingUtilsMigration extends AbstractMigration {
+    private static final Logger logger = LoggerFactory.getLogger(ReplaceRegexMatchUsingUtilsMigration.class);
+    private static final String JAVA_FILE_REGEX = "JavaFunctionToBeMigrated";
+    private static final String NEW_JAVA_FILE_NAME = "MigratedJavaFunction";
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        boolean shouldExecuteMigration = false;
+        try {
+            logger.info("Detecting migrations for {} to {}", JAVA_FILE_REGEX, NEW_JAVA_FILE_NAME);
+            shouldExecuteMigration = FileUtils.hasRegExMatch(JAVA_FILE_REGEX, file);
+        } catch (Exception e) {
+            logger.error("Caught exception checking Java files for migration legibility", e);
```

Comment:
S: I'm still a newbie to Baton but it appears there's a BatonException class, would it be better for consistency to incorporate that special exception logic to handle your exceptions?

Reply:
Yep, good catch. For consistency, this now catches an IOException, logs an error, and throws the exception as a BatonException.